### ISSUE TITLE
Feature: deal with huge result sets

### ISSUE THREAD

**Issue by geoffharcourt** [thoughtbot/administrate#430]
We have an app with tens of thousands of people in it. There are several records that have a person as an association. On the form page for a model that belongs to a person, trying to populate a select list of people results in the app timing out on Heroku or choking when the JS-enabled select field is loaded with all the possible values.

A widget with a search functionality if the potential result set is larger than 10 or so items would be awesome. Administrate already has a search functionality, so this could piggyback on that API.

**Comment by graysonwright** [comment]
Agreed, I started to look into this a little bit earlier, but punted until later. I'm going to add this to the "flexibility" milestone, 'cuz I'd like to see this on the roadmap, but I think extensibility takes priority.

If anyone wants to take a crack at this, have at it!

**Comment by igorbozato** [comment]
Is anyone working on this one? I would like to work on it.

**Comment by nickcharlton** [comment]
Hi @igorbozato! Please do! I think it'd be interesting to see a solution that only involved JavaScript on really huge result sets (so we don't load things which aren't needed) and so forth.
